### PR TITLE
feat(coordinaterequest): Convert convenience functions to echo from vk.ctx

### PR DESCRIPTION
**Pull request by javorszky**
Relates to #35 

Really though, coordinatedRequest should not depend on any one implementation of a routing library.

This is a temporary stopgap.

### CODE REVIEW

**Review by rnpridgeon — APPROVED** [review]

**Comment by rnpridgeon** [comment]
>  coordinatedRequest should not depend on any one implementation of a routing library.

agreed

File: request/request.go [hunk 1]
```diff
@@ -30,59 +30,59 @@ type CoordinatedRequest struct {
...
+	bodyValues map[string]interface{}
 }
 
-// FromVKRequest creates a CoordinatedRequest from an VK request handler
-func FromVKRequest(r *http.Request, ctx *vk.Ctx) (*CoordinatedRequest, error) {
-	reqBody, err := io.ReadAll(r.Body)
-	if err != nil {
-		return nil, vk.E(http.StatusInternalServerError, "failed to read request body")
+// FromEchoContext creates a CoordinatedRequest from an echo context.
+func FromEchoContext(c echo.Context) (*CoordinatedRequest, error) {
```
Comment (rnpridgeon) [inline request/request.go:37]: Make sure to link the corresponding e2core PR when you are done just so its easy to spot when looking at the commit history.

**Comment by javorszky** [comment]
There was also a part where an options interface was used to make sure the sources have loggers.

I removed it. The system spec should not prescribe that a particular logger needs to be used, nor should it enforce that there needs to be a logger to begin with. Each implementation can decide whether they include a logger, and which one that is.

**Comment by rnpridgeon** [comment]
also agreed, still LGTM

**Pull request merged**